mobify/platform-scaffold
# Add ESlint rule to uncomment Merlin's Connector 🎩

# Adds ESlint rule to uncomment Merlin's Connector 🎩

This PR adds a linting error if any connector is left uncommented other than Merlin's, as it should remain the "enabled-by-default" connector.

This plugin and the associated `package.json` import and `.eslintrc` rule should perhaps be removed by the `platform-generator` when a new project is generated.

## How to test-drive this PR
```bash
git checkout add-merlins-connector-eslint-rule
npm i
```

- Change to the SFCC connector
- Go into your editor and disable -> re-enable eslint (restarts eslint server), then check `main.jsx` for errors
- Or run `npm run lint`

<img width="852" alt="screen shot 2017-06-07 at 3 09 43 pm" src="https://user-images.githubusercontent.com/16109837/26903730-614cb048-4b93-11e7-84f0-598359845d31.png">

<img width="727" alt="screen shot 2017-06-07 at 3 11 19 pm" src="https://user-images.githubusercontent.com/16109837/26903755-93aaf2d4-4b93-11e7-8a43-cb0cb88112a5.png">

## Resources

ESlint uses an abstract syntax tree (AST) to navigate JavaScript code, to see what the AST looks like for any set of code, paste it into the sandbox editor in the link below
- http://astexplorer.net/
